### PR TITLE
x86: Fix zefi.py creating valid images

### LINKED ISSUE #31517: UP² broken (git bisect findings inside)
**Describe the bug**
Any samples running on UP² that has a console output should work, but it does not anymore.

**To Reproduce**

I tried the tests/arch/x86/info sample
up_squared board  (so x86_64) on multiboot (zephyr.strip)

**Expected behavior**
A lot of output on the multiboot, acpi and RTC

**Impact**
I cannot test my work on UP² atm, and I am using it extensively

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain:Zephyr SDK
 - Commit SHA: ef1b4d45b0ae1fe3cfd9e85d45acdf595a23a3b1

**Additional context**

I ran a git bisect, and it finds the commit 5c47bbc5010d98c6f6d795aa6d268138c3463bbd

It previously was commit 3660040e22042e0e67005da0a7229fc224e089d0 which was the same before it got reverted.

### PR DESCRIPTION
**Pull request by galak**
When zefi.py was changed to pass compiler and objcopy the flag to
objcopy for the EFI target was dropped.  This is because the current
SDK (0.12.1) doesn't support that target type for objcopy.  However,
target is necessary for the images to be created correctly and boot.

Switch back to use the host objcopy as a stop gap fix, until the SDK
can support target for EFI.

Fixes: #31517

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]
Verified to work with ehl_crb

**Comment by galak** [comment]
https://github.com/zephyrproject-rtos/sdk-ng/issues/318 to add the support in the binutils build for x86_64.

**Comment by galak** [comment]
Sent a PR for the SDK to enable the EFI binary support in binutils.  Once we have some test SDK build we can try that out.

File: boards/x86/ehl_crb/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ if(CONFIG_BOARD_EHL_CRB)
 set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
   COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/arch/x86/zefi/zefi.py
   -c ${CMAKE_C_COMPILER}
-  -o ${CMAKE_OBJCOPY}
+  -o objcopy
```
Comment (nashif) [inline boards/x86/ehl_crb/CMakeLists.txt:6]: will this now use host objcopy? will this work across architectures?

Comment (galak) [reply boards/x86/ehl_crb/CMakeLists.txt:6]: No, its a stop gap until the SDK fix goes in.

I've got a SDK enhancement to enable EFI binary generation in the x86_64 toolchain.  So this is a temp fix, until we put out SDK 0.12.2 which I will do once we get a test SDK build and can confirm it works.

**Review by nashif — APPROVED** [review]

**Review by yerabolu — APPROVED** [review]

**Review by jenmwms — APPROVED** [review]

**Pull request merged**